stdlib/messaging/kafka/src/main/java/org/ballerinalang/messaging/kafka/observability/KafkaMetricsUtil.java

```diff
@@ -0,0 +1,364 @@
+        if (!ObserveUtils.isMetricsEnabled()) {
+            return;
+        }
+        KafkaObserverContext observerContext = new KafkaObserverContext(KafkaObservabilityConstants.CONTEXT_PRODUCER,
+                                                                        KafkaUtils.getClientId(producerObject),
+                                                                        KafkaUtils.getBootstrapServers(producerObject),
+                                                                        topic);
+        reportPublish(observerContext, size);
+    }
+
+    /**
+     * Reports a consumer subscribing to a subject.
+     *
+     * @param consumerObject    Consumer object.
+     * @param topicsList        List of topics that the consumer subscribes to.
+     */
+    public static void reportBulkSubscription(ObjectValue consumerObject, Set<String> topicsList) {
+        if (!ObserveUtils.isMetricsEnabled()) {
+            return;
+        }
+        for (String topic : topicsList) {
+            reportSubscription(consumerObject, topic);
+        }
+    }
+
+
+    /**
+     * Reports a consumer subscribing to a topic.
+     *
+     * @param consumerObject    Consumer object.
+     * @param topic             Topic that the consumer subscribes to.
+     */
+    public static void reportSubscription(ObjectValue consumerObject, String topic) {
+        if (!ObserveUtils.isMetricsEnabled()) {
+            return;
+        }
+        KafkaObserverContext observerContext = new KafkaObserverContext(KafkaObservabilityConstants.CONTEXT_CONSUMER,
+                                                                        KafkaUtils.getClientId(consumerObject),
+                                                                        KafkaUtils.getBootstrapServers(consumerObject),
+                                                                        topic);
+        setGauge(observerContext, KafkaObservabilityConstants.METRIC_SUBSCRIPTION[0],
+                       KafkaObservabilityConstants.METRIC_SUBSCRIPTION[1], 1);
+    }
+
+    /**
+     * Reports a consumer unsubscribing from a subject.
+     *
+     * @param consumerObject    Consumer object.
+     * @param topic             Subject that the consumer unsubscribes from.
+     */
+    public static void reportUnsubscription(ObjectValue consumerObject, String topic) {
+        if (!ObserveUtils.isMetricsEnabled()) {
+            return;
+        }
+        KafkaObserverContext observerContext = new KafkaObserverContext(KafkaObservabilityConstants.CONTEXT_CONSUMER,
+                                                                        KafkaUtils.getClientId(consumerObject),
+                                                                        KafkaUtils.getBootstrapServers(consumerObject),
+                                                                        topic);
+        setGauge(observerContext, KafkaObservabilityConstants.METRIC_SUBSCRIPTION[0],
```

Comment:
Shall we created separate method for `unsetGauge` rather having a integer parameter to distinguish the logic?
It is more clearer and consistent with other methods.